@here/harp-mapview/lib/Tile.ts

```diff
@@ -909,6 +887,14 @@ export class Tile implements CachedResource {
             return await Promise.resolve();
         }
 
+        if (this.m_tileGeometryLoader) {
```

Comment:
Minor: why not move this into the then below, that means, if the tile is disposed before it is loaded, that this extra callback isn't queued.

Reply:
The tile might be disposed before the decoding finishes. Then, the TileGeometryLoader rejects its promise and that'll cause and exception cause there's no reject callback attached.
That's actually the issue we have right now in master.

Reply:
I got ya.

You could alternatively in that use `finally`, so that even if it is disposed, that the reset method is called. But not super important, makes it a bit easier to read, but also ok.